functions/transferChatStart.ts

```diff
@@ -93,6 +100,81 @@ async function closeTaskAndKick(context: Context<EnvVars>, body: Required<Body>)
   return closedTask;
 }
 
+// if transfer targets a worker, validates that it can be effectively transferred, and if the worker's chat capacity needs to increase
+async function validateChannelIfWorker(
+  context: Context<EnvVars>,
+  targetSid: string,
+  transferTargetType: string,
+  taskChannelUniqueName: string,
+  channelType: string,
+) {
+  if (transferTargetType === 'queue') return { type: 'queue' } as const;
+
+  const client = context.getTwilioClient();
+
+  const [worker, workerChannel] = await Promise.all([
+    client.taskrouter
+      .workspaces(context.TWILIO_WORKSPACE_SID)
+      .workers(targetSid)
+      .fetch(),
+    client.taskrouter
+      .workspaces(context.TWILIO_WORKSPACE_SID)
+      .workers(targetSid)
+      .workerChannels(taskChannelUniqueName)
+      .fetch(),
+  ]);
+
+  if (!worker.available)
+    return {
+      type: 'error',
+      payload: { status: 403, message: "Error: can't transfer to an offline counselor" },
+    } as const;
+
+  const workerAttr = JSON.parse(worker.attributes);
+
+  const unavailableVoice = channelType === 'voice' && !workerChannel.availableCapacityPercentage;
+  const unavailableChat =
+    channelType !== 'voice' &&
+    !workerChannel.availableCapacityPercentage &&
+    workerAttr.maxMessageCapacity &&
```

Comment:
I think this function needs a test for when `maxMessageCapacity` has not been set.  It is possible that whoever is configuring the system could overlook setting it for a particular worker.  I think a reasonable default in that case should be to rely only on the configured capacity as is.  But here I think `unavailableChat` will be `false` if `maxMessageCapacity` is `null`, which would be misleading.

Reply:
Addressed in https://github.com/tech-matters/serverless/pull/25/commits/b3ede0c809d8df1b0c04a8aa94830b1941bae5ef